### PR TITLE
Run migration logic with leader election

### PR DESCRIPTION
**Pull request by rollandf**
- Add Controller Migrator to the Manager
- Add a channel to block reconcile until Migrator finishes

### CODE REVIEW

File: pkg/ipam-controller/controllers/ippool/ippool.go [hunk 1]
```diff
@@ -46,13 +46,15 @@ import (
...
 	NodeEventCh    chan event.GenericEvent
+	MigrationCh    chan struct{}
 	client.Client
 	Scheme   *runtime.Scheme
 	recorder record.EventRecorder
 }
 
 // Reconcile contains logic to sync IPPool objects
 func (r *IPPoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	<-r.MigrationCh
```
Comment (ykulazhenkov) [inline pkg/ipam-controller/controllers/ippool/ippool.go:57]: do we want to use 
```
select {
        case <-r.MigrationCh: 
        case <- ctx.Done():
           return ctrl.Result{}, fmt.Errorf("canceled")
        }

```
or something similar to support clean shutdown?

Comment (rollandf) [reply pkg/ipam-controller/controllers/ippool/ippool.go]: Done

**Review by ykulazhenkov — DISMISSED** [review]

File: pkg/ipam-controller/migrator/migrator.go [hunk 2]
```diff
@@ -50,10 +50,35 @@ const (
...
+type Migrator struct {
+	K8sClient        client.Client
+	IPPoolsNamespace string
+	MigrationCh      chan struct{}
+	LeaderElection   bool
+	Logger           logr.Logger
+}
+
+// Implements manager.Runnable
+func (m Migrator) Start(ctx context.Context) error {
```
Comment (ykulazhenkov) [inline pkg/ipam-controller/migrator/migrator.go:63]: do we want to use pointer receiver here? (m *Migrator)

**Review by ykulazhenkov — COMMENTED** [review]
one more comment to address before merging

**Review by ykulazhenkov — APPROVED** [review]
LGTM

**Pull request merged**